lib/engine/action/base.rb

```diff
@@ -8,12 +8,13 @@ class Base
       include Helper::Type
 
       attr_reader :entity
-      attr_accessor :id, :user
+      attr_accessor :id, :user, :created_at
 
       def self.from_h(h, game)
         entity = game.get(h['entity_type'], h['entity']) || Player.new(nil, h['entity'])
         obj = new(entity, **h_to_args(h, game))
         obj.user = h['user'] if entity.player && h['user'] != entity.player&.id
+        obj.created_at = h['created_at']
```

Comment:
would it make sense to have the || Time.now here?

Reply:
Yes. Done and done. Also in initialize. Had to change all the action class to call super during initialize too. Surprised they weren't already.